packages/keychain/src/pages/slot/session/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { Policy, Session } from "@cartridge/controller";
+import { LoadingLogo } from "components";
+import { NextPage } from "next";
+import Controller from "utils/controller";
+import { CreateSession } from "components/connect";
+
+import { diff } from "utils/controller";
+import { fetchAccount } from "components/connect/utils";
+import { useConnection } from "hooks/connection";
+import { useRouter } from "next/router";
+import { useCallback, useEffect, useState } from "react";
+
+type SessionQueryParams = Record<string, string> & {
+  callback_uri: string;
+  username: string;
+};
+
+/**
+    This page is for creating session with Slot
+*/
+const CreateSessionPage: NextPage = () => {
+  const router = useRouter();
+  const queries = router.query as SessionQueryParams;
+
+  const { controller, setController, policies, origin, chainId, rpcUrl } =
+    useConnection();
+
+  // Fetching account status for displaying the loading screen
+  const [isFetching, setIsFetching] = useState(false);
+
+  // Handler for calling the Slot callback uri.
+  // Send the session details to the callback uri in the body of the
+  // POST request. If the request is successful, then redirect to the
+  // success page. Else, redirect to the failure page.
+  const onSlotCallback = useCallback(
+    (session: Session) => {
+      const url = decodeURIComponent(queries.callback_uri);
+      const body = JSON.stringify(session);
+      const headers = new Headers();
+      headers.append("Content-Type", "application/json");
+
+      fetch(url, {
+        body,
+        headers,
+        method: "POST",
+      })
+        .then((res) => {
+          res.status === 200
+            ? router.replace(`/slot/auth/success`)
+            : new Promise((_, reject) => reject(res));
```

Comment:
I think we need to `throw new Error()` here, otherwise this is uncaught

Reply:
i believe it should be caught with chain'd '.catch()`, no?

Reply:
Mmm if u try to make it fail, there's an uncaught error message. You could also do `return Promise.reject()`

Reply:
ok i just have to make it a return statement, fixed it https://github.com/cartridge-gg/controller/pull/369/commits/bb3b07eee519d713d3260dc59baf6d29dc8a68e9 .